lib/definition_context.rb

```diff
@@ -0,0 +1,34 @@
+class DefinitionContext
+  def initialize(deployment, container_hostname)
+    @deployment = deployment
+    @container_hostname = container_hostname
+  end
+
+  def passthrough(value: )
+    value
+  end
+
+  def deployment_ip_addresses(deployment_name: )
+    containers = Deployment.find_by(name: deployment_name).wanted_existing_containers
+    containers.pluck(:ipaddress).compact
+  end
+
+  def container_id
+    parse_container_id(@container_hostname)
+  end
+
+  def deployment_host_sequences(host: , self_host: "0.0.0.0")
+    @deployment.container_names.map do |name|
+      if @container_hostname == name
+        self_host
+      else
+        "#{parse_container_id(name)}.#{host}"
+      end
+    end
+  end
+
+  private
+  def parse_container_id(hostname)
+    hostname[(@deployment.name.length + 1)..-1].to_i
```

Comment:
can you elaborate a bit, why do we have to do this?
this method is not intuitive, either we improve the clarity or add comments

Reply:
Previously we agreed that containers managed by a deployment must have this name: **deployment name-dash-number.**

This method is needed to get the container number, which is needed by Zookeeper `my_id` and `zookeeper_host` bootstrap attribute.

Reply:
ok got it, just realized it